From 260d4b742bc36744aa2282421547f1a7b483d2f8 Mon Sep 17 00:00:00 2001
From: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date: Mon, 3 Apr 2023 14:18:14 -0300
Subject: [PATCH 26/27] nptl: Fix tst-cancel30 on sparc64

As indicated by sparc kernel-features.h, even though sparc64 defines
__NR_pause,  it is not supported (ENOSYS).  Always use ppoll or the
64 bit time_t variant instead.

(cherry picked from commit 370da8a121c3ba9eeb2f13da15fc0f21f4136b25)
---
 sysdeps/pthread/tst-cancel30.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/sysdeps/pthread/tst-cancel30.c b/sysdeps/pthread/tst-cancel30.c
index 6eb4eb5b42..ff803386be 100644
--- a/sysdeps/pthread/tst-cancel30.c
+++ b/sysdeps/pthread/tst-cancel30.c
@@ -46,9 +46,7 @@ tf (void *arg)
 
   /* Wait indefinitely for cancellation, which only works if asynchronous
      cancellation is enabled.  */
-#ifdef SYS_pause
-  syscall (SYS_pause);
-#elif defined SYS_ppoll || defined SYS_ppoll_time64
+#if defined SYS_ppoll || defined SYS_ppoll_time64
 # ifndef SYS_ppoll_time64
 #  define SYS_ppoll_time64 SYS_ppoll
 # endif
-- 
2.41.0

